File: HoverDome_OC/HoverDome_OC/Children1ViewController.m


#import "Children1ViewController.h"

@interface Children1ViewController ()<UITableViewDataSource,UITableViewDelegate>
@property(nonatomic, strong) UITableView *tableView;
@end

@implementation Children1ViewController

- (void)setOffsetY:(CGFloat)offsetY{
    self.tableView.contentOffset = CGPointMake(0, offsetY);
}

- (CGFloat)offsetY{
    return self.tableView.contentOffset.y;
}

- (void)setIsCanScroll:(BOOL)isCanScroll{
    if (isCanScroll == YES){
        [self.tableView setContentOffset:CGPointMake(0, self.offsetY) animated:NO];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    _tableView = [[UITableView alloc]initWithFrame:self.view.bounds style:UITableViewStylePlain];
    [_tableView registerClass: [UITableViewCell class] forCellReuseIdentifier:@"123"];
    _tableView.dataSource = self;
    _tableView.delegate = self;
    [self.view addSubview:_tableView];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 100;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"123" forIndexPath:indexPath];
    cell.textLabel.text = [NSString stringWithFormat:@"%zd", indexPath.row];
    return cell;
}

- (void)scrollViewDidScroll:(UIScrollView *)scrollView{
    if ([self.scrollDelegate respondsToSelector:@selector(hoverChildViewController:scrollViewDidScroll:)]){
        [self.scrollDelegate hoverChildViewController:self scrollViewDidScroll:scrollView];
    }
}

@end

File: HoverDome_Swift/HoverDome/Children1ViewController.swift
//
//  Children1ViewController.swift
//  HoverDome
//
//  Created by admin on 2019/2/27.
//  Copyright © 2019 zhouqiao. All rights reserved.
//

import UIKit

class Children1ViewController: HoverChildViewController {

    private var tableView:UITableView!
    private lazy var items: [String] = {
        var items = [String]()
        for i in 0..<100 {
            items.append("\(i)")
        }
        return items
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.green
        
        tableView = UITableView(frame: view.bounds, style: .plain)
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "aaaa")
        tableView.delegate = self
        tableView.dataSource = self
        view.addSubview(tableView)
        scrollView = tableView
    }

    override var offsetY: CGFloat{
        set{
            tableView.contentOffset = CGPoint(x: 0, y: newValue)
        }
        get{
            return tableView.contentOffset.y
        }
    }
    
    override var isCanScroll: Bool{
        didSet{
            if isCanScroll{
                tableView.setContentOffset(CGPoint(x: 0, y: offsetY), animated: false)
            }
        }
    }
}

extension Children1ViewController: UITableViewDataSource, UITableViewDelegate {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return items.count
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "aaaa", for: indexPath)
        cell.textLabel?.text = items[indexPath.row]
        return cell
    }

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        let vc = UIViewController()
        navigationController?.pushViewController(vc, animated: true)
        
    }

    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        scrollDelegate?.hoverChildViewController(self, scrollViewDidScroll: scrollView)
    }
}
